### PR TITLE
New module to set up cloud object

### PR DESCRIPTION
**Pull request by grastogi23**
##### ISSUE TYPE

 - New Module Pull Request

##### COMPONENT NAME

avi.avi_cloud

##### ANSIBLE VERSION

```
ansible 2.4.0
  config file = /home/grastogi/.ansible.cfg
  configured module search path = [u'/home/grastogi/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/grastogi/ansible2.3/local/lib/python2.7/site-packages/ansible
  executable location = /home/grastogi/ansible2.3/bin/ansible
  python version = 2.7.6 (default, Jun 22 2015, 17:58:13) [GCC 4.8.2]
```

##### SUMMARY

New module to setup Avi Cloud object

```

```
#

### CODE REVIEW

**Comment by mattclay** [comment]
CI failure due to module docs validation error:

```
2017-03-02 15:10:40 ============================================================================
2017-03-02 15:10:40 lib/ansible/modules/network/avi/avi_cloud.py
2017-03-02 15:10:40 ============================================================================
2017-03-02 15:10:40 TRACE:
2017-03-02 15:10:40     while parsing a block collection
2017-03-02 15:10:40       in "<string>", line 48, column 13:
2017-03-02 15:10:40                     - Ipam profile for east-west ser ... 
2017-03-02 15:10:40                     ^
2017-03-02 15:10:40     expected <block end>, but found '<scalar>'
2017-03-02 15:10:40       in "avi_cloud.DOCUMENTATION", line 76, column 25:
2017-03-02 15:10:40                     - [warning] please use virtual subnets in th ... 
2017-03-02 15:10:40                                 ^
2017-03-02 15:10:40 ERROR:302:DOCUMENTATION is not valid YAML. Line 76 column 25
```

**Comment by khaltore** [comment]
shipit

**Comment by ericsysmin** [comment]
shipit

**Comment by gundalow** [comment]
Closing and reopening to retrigger CI as we've extended module validation rules.

File: lib/ansible/modules/network/avi/avi_cloud.py [hunk 1]
```diff
@@ -0,0 +1,243 @@
...
+            - CLOUD_RANCHER, CLOUD_OSHIFT_K8S.
+            - Default value when not specified in API or module is interpreted by Avi Controller as CLOUD_NONE.
+        required: true
+extends_documentation_fragment:
+    - avi
+'''
+
+
+EXAMPLES = '''
+  - name: Create a VMWare cloud with write access mode
```
Comment (gundalow) [inline lib/ansible/modules/network/avi/avi_cloud.py:161]: s/VMWare/VMware/g

**Review by gundalow — APPROVED** [review]
Only a minor comment, happy for you to update any instances of this in a follow up PR across all modules

File: lib/ansible/modules/network/avi/avi_cloud.py (hunk 1 above)
Comment (gundalow) [reply lib/ansible/modules/network/avi/avi_cloud.py:161]: I'm happy for you to raise one PR that fixes that across any/all modules it affects at a later point.

Comment (grastogi23) [reply lib/ansible/modules/network/avi/avi_cloud.py:161]: I will fix them.

**Pull request merged**